server/server.go

```diff
@@ -576,14 +587,19 @@ func (s *server) CreateSubnets(ctx context.Context, req *rpcpb.CreateSubnetsRequ
 	// to decide cluster/subnet readiness
 	go func() {
 		s.waitChAndUpdateClusterInfo("waiting for custom chains readiness", readyCh, true)
+		s.mu.Unlock()
 	}()
 
-	return &rpcpb.CreateSubnetsResponse{ClusterInfo: s.clusterInfo}, nil
+	return &rpcpb.CreateSubnetsResponse{ClusterInfo: info}, nil
 }
 
 func (s *server) Health(ctx context.Context, req *rpcpb.HealthRequest) (*rpcpb.HealthResponse, error) {
 	s.log.Debug("health")
-	if info := s.getClusterInfo(); info == nil {
+	s.mu.Lock()
```

Comment:
I am now confused about what `Health` is supposed to be doing. `Health` was at some point just a call which was returning right away, with the information about if the network is healthy or not. 

Now it seems to be blocking, which in my opinion is strange. What if a client wants just to ask if the network is healthy, without waiting? 

Then, we can have a blocking function `WaitUntilHealthy` which is doing just that. That is much clearer in the wording I think. Also, there is no documentation in the code about what these functions are doing. The documentation should say things like if it is blocking, non blocking, if it assumes a lock held etc. Otherwise it's very difficult for users of these functions to understand the code.